test/firtool/type-alias.fir

```diff
@@ -0,0 +1,28 @@
+; RUN: firtool %s --preserve-aggregate=all --scalarize-top-module=false -ir-fir | FileCheck %s
+
+FIRRTL version 3.0.0
```

Comment:
What's this test for, is this checking more than parsing (`parse-basic`)?

Reply:
I meant this to be e2e tests of type alias whereas parse-basic is parser only. Will mention that, thanks!

Reply:
Works for me! Thanks!